docs: update demo docs for user registration (#721)
ZcashFoundation/frost: book/src/zcash/ywallet-demo.md

RUST_LOG=debug cargo run --bin server
```

### Registering users

In order to interact with the server, you will need to register users. For this
guide we will need two. In a new terminal, run the following command for user
"alice" (replace the password if you want):

```

> curl --data-binary '{"username": "alice", "password": "foobar10", "pubkey": ""}' -H 'Content-Type: application/json' http://127.0.0.1:2744/register

Reviewer: MIght be good to explain what the pubkey parameter is about and why it's an empty string here

Author: Done in 412dd015

```

It will output "null". Also register user "bob":

```
curl --data-binary '{"username": "bob", "password": "foobar10", "pubkey": ""}' -H 'Content-Type: application/json' http://127.0.0.1:2744/register
```

> You only need to do this once, in case you want to sign more than one

Reviewer: This sentence is a bit confusing

Author: Fixed in 412dd015

> You only need to do this once, in case you want to sign more than one

Reviewer: ```suggestion
You only need to do this once, even if you want to sign more than one
```

Author: Fixed in 412dd015

transaction. If for some reason you want to start over, close the server and
delete the `db.sqlite` file.

Feel free to close this terminal, or reuse it for the next step.

```admonish warning
Do not use passwords that you use in practice; use dummy ones instead. (You
shouldn't reuse passwords anyway!) For real world usage you would need to take
more care to not end up writing the password to your shell history. (In real
world usage we'd expect this to be done by applications anyway.)
```

### Coordinator

In the second terminal, the Coordinator, run (in the same folder where key
generation was run):

```
export PW=foobar10
cargo run --bin coordinator -- -C redpallas --http -u alice -w PW -S alice,bob -r -
```

We will use "alice" as the Coordinator, so change the value next to `export PW=`
if you used another password when registering "alice".

And then:

- It should read the public key package from `public-key-package.json`.

> - When prompted for the message to be signed, paste the SIGHASH printed by the

Reviewer: Here it's waiting for participants to send their commitments first, maybe we should add this before mentioning the prompt for the message to be signed

Author: Ah I see, I ended up using https://github.com/ZcashFoundation/frost-zcash-demo/pull/265 to write these docs, and the flow changed there. Since that's already approved I think it's OK, we can merge the docs after both that and the user registration PR merge.

argument, you will need to convert it to binary format.
```

### Participant 1 (alice)

In the third terminal, Participant 1, run the following:

```
export PW=foobar10
cargo run --bin participant -- -C redpallas --http --key-package key-package-1.json -u alice -w PW
```

(We are using "alice" again. There's nothing stopping a Coordinator from being a
Partcipant too!)

### Participant 2 (bob)

In the fourth terminal, for Participant 2, run the following:

```

> cargo run --bin participant -- -C redpallas --http --key-package key-package-2.json -u bob -w PW

Reviewer: Here it prompted me to set the password. Should we add `export PW=foobar10`  to this command also or will it depend on how people have configured things in their terminals?

Author: Good catch, fixed in 412dd015
